mapzen-places-api/src/main/java/com/mapzen/places/api/internal/PlaceAutocompleteController.java

```diff
@@ -27,4 +28,10 @@
    * @return
    */
   LatLngBounds getBounds();
+
+  /**
+   * Return the filter that should be used to limit autocomplete results.
+   * @return
+   */
+  AutocompleteFilter getAutocompleteFilter();
```

Comment:
This is kind of semantic point, but typically in MVP the Presenter does not "get" anything from the View/Controller. But rather the View/Controller pushes data to the Presenter when it is received as input then acts as a listener waiting for a callback with further instructions.

Rather than having two methods `getBounds()` and `getAutocompleteFilter()` here a more idiomatic way to achieve the same result would be to parse the incoming `Intent` extras in `PlaceAutocompleteActivity.onCreate()` and set the `LatLngBounds` and `AutocompleteFilter` on the `PlaceAutocompletePresenter` as soon as they are available to be stored in a field until needed.

This also has the added benefit of making the Presenter easier to test as you can set these values from the test method rather than registering a stub View/Controller to supply the values on demand.